kwok/ec2/ec2.go

```diff
@@ -0,0 +1,890 @@
+	}
+	log.FromContext(ctx).WithValues("count", total).Info("loaded instances from backup")
+}
+
+//nolint:gocyclo
+func (c *Client) backupInstances(ctx context.Context) error {
+	var instances []ec2types.Instance
+	c.instances.Range(func(k, v interface{}) bool {
+		instances = append(instances, v.(ec2types.Instance))
+		return true
+	})
+	sort.Slice(instances, func(i, j int) bool {
+		return lo.FromPtr(instances[i].LaunchTime).Before(lo.FromPtr(instances[j].LaunchTime))
+	})
+
+	configMaps := &corev1.ConfigMapList{}
+	if err := c.kubeClient.List(ctx, configMaps, client.InNamespace("kube-system")); err != nil {
+		return fmt.Errorf("listing configmaps, %w", err)
+	}
+	configMaps.Items = lo.Filter(configMaps.Items, func(c corev1.ConfigMap, _ int) bool {
+		return strings.Contains(c.Name, "kwok-aws-instances-")
+	})
+	sort.SliceStable(configMaps.Items, func(i, j int) bool {
+		rawI := strings.Split(configMaps.Items[i].Name, "kwok-aws-instances-")
+		if len(rawI) != 2 {
+			return false
+		}
+		rawJ := strings.Split(configMaps.Items[j].Name, "kwok-aws-instances-")
+		if len(rawJ) != 2 {
+			return false
+		}
+		iNum, err := strconv.Atoi(rawI[1])
+		if err != nil {
+			return false
+		}
+		jNum, err := strconv.Atoi(rawJ[1])
+		if err != nil {
+			return false
+		}
+		return iNum < jNum
+	})
+	// Clean-up any ConfigMaps that don't need to be there because of the count
+	// We store 500 instances per ConfigMap
+	numConfigMaps := int(math.Ceil(float64(len(instances)) / float64(500)))
+	if numConfigMaps < len(configMaps.Items) {
+		errs := make([]error, numConfigMaps)
+		workqueue.ParallelizeUntil(ctx, 10, len(configMaps.Items)-numConfigMaps, func(i int) {
+			if err := c.kubeClient.Delete(ctx, &configMaps.Items[len(configMaps.Items)-i-1]); client.IgnoreNotFound(err) != nil {
+				errs[i] = fmt.Errorf("deleting configmap %q, %w", configMaps.Items[len(configMaps.Items)-i-1].Name, err)
+			}
+		})
+		if err := multierr.Combine(errs...); err != nil {
+			return err
+		}
+	}
+
+	errs := make([]error, numConfigMaps)
+	workqueue.ParallelizeUntil(ctx, 10, numConfigMaps, func(i int) {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("kwok-aws-instances-%d", i),
+				Namespace: "kube-system",
+			},
+		}
+		if err := c.kubeClient.Get(ctx, client.ObjectKeyFromObject(cm), cm); err != nil {
+			if k8serrors.IsNotFound(err) {
+				if err = c.kubeClient.Create(ctx, cm); err != nil {
+					errs[i] = fmt.Errorf("creating configmap %q, %w", cm.Name, err)
+					return
+				}
+			} else {
+				errs[i] = fmt.Errorf("getting configmap %q, %w", cm.Name, err)
+				return
+			}
+		}
+		stored := cm.DeepCopy()
+		cm.Data = map[string]string{"instances": string(removeNullFields(lo.Must(json.Marshal(lo.Slice(instances, i*500, (i+1)*500)))))}
+		if !equality.Semantic.DeepEqual(cm, stored) {
+			if err := c.kubeClient.Patch(ctx, cm, client.MergeFrom(stored)); err != nil {
+				errs[i] = fmt.Errorf("patching configmap %q, %w", cm.Name, err)
```

Comment:
No retry?

Reply:
Yeah, the idea is that when we return an error, we will just continue and immediately go back into this function to retry